sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScanBase.scala

```diff
@@ -871,6 +872,28 @@ trait ParquetPartitionReaderBase extends Logging with Arm with ScanWithMetrics
     currentChunk
   }
 
+  /**
+   * Take case-sensitive into consideration when getting the data reading column names
+   * before sending parquet-format buffer to cudf.
+   *
+   * @param readDataSchema spark schema to read
+   * @param fileSchema  the schema of the dumped parquet-format buffer
+   * @param isCaseSensitive if it is case sensitive
+   * @return a sequence of column names following the order of readDataSchema
+   */
+  protected def toCudfColumnNames(
+      readDataSchema: StructType,
+      fileSchema: MessageType,
+      isCaseSensitive: Boolean): Seq[String] = {
+    val fields = fileSchema.getFields.asScala.map(_.getName).toSet
+
+    if (!isCaseSensitive) {
+      val m = CaseInsensitiveMap(fields.zip(fields).toMap)
+      readDataSchema.fieldNames.map(m(_))
```

Comment:
Test cases are failing for missing columns, this will need to gracefully handle the case where lookup fails.